test/hotspot/gtest/aarch64/test_assembler_aarch64.cpp

```diff
@@ -81,4 +82,203 @@ TEST_VM(AssemblerAArch64, validate) {
   BufferBlob::free(b);
 }
 
+static void asm_dump(address start, address end) {
+  ResourceMark rm;
+  stringStream ss;
+  ss.print_cr("Insns:");
+  Disassembler::decode(start, end, &ss);
+  printf("%s\n", ss.as_string());
+}
+
+TEST_VM(AssemblerAArch64, merge_dmb) {
```

Comment:
Given the previous experience with barrier merges that prompted the backout, I would prefer to have a more comprehensive test here, maybe an additional one. I am thinking something like the exhaustive combination of 4 back-to-back barriers of each of 5 types. This gives us 5^4 = 625 test cases, which I think is still manageable.

Reply:
Test is added as merge_dmb_all_kinds

Reply:
Right. I was implicitly thinking that we can do this without coding the explicit patterns into the test. As it stands now, it is hard to check that generated patterns are actually correct. Let me see if I can whip up a sample of what I had in mind.

Reply:
I was thinking about this: [improve-tests.patch](https://github.com/openjdk/jdk/files/15419452/improve-tests.patch)